source/MRViewer/MRSurfaceManipulationWidget.cpp

```diff
@@ -190,6 +198,13 @@ void SurfaceManipulationWidget::changeSurface_()
         const float r = std::clamp( distances_[v] / settings_.radius, 0.f, 1.f );
         const float k = r < intensity ? a1 * r * r + 1 : a2 * ( r - 1 ) * ( r - 1 ); // I(r)
         float pointShift = maxShift * k; // shift = F * I(r)
+        if ( pointShift > pointsShift_[v] )
+        {
+            pointShift -= pointsShift_[v];
+            pointsShift_[v] += pointShift;
```

Comment:
```suggestion
            pointsShift_[v] = pointShift;

Reply:
if ( a > b )
{
    c = a - b;
    d = b + c; ( d = b + a - b = a  )
}

Reply:
yes, but we remember b(d) as a, and we know difference a-b (c)